src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -594,11 +603,119 @@ def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
+        # groupby_level will be index level to group by (the first level index of member_ids (child_id or mother_id).
         member_ids = (
             member_ids.reset_index()
-            .groupby(["child_id"])["person_id"]
+            .groupby([groupby_level])["person_id"]
             .apply(np.random.choice)
         )
         return member_ids
+
+    def assign_second_guardian_to_newborns(
+        self, new_births: pd.DataFrame, households: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        new_births: pd.DataFrame of new births on this time step
+        households: pd.DataFrame of the entire state table for the two columns of household_id
+          and relation_to_household_head.
+
+        Returns
+        -------
+        new_births with the additional guardian_2 column updated.
+        """
+        # Setup
+        new_births["guardian_2"] = data_values.UNKNOWN_GUARDIAN_IDX
+        mothers_households = self.get_mothers_household_structure(
+            new_births["parent_id"], households
+        )
+        partners = [
+            "Opp-sex spouse",
+            "Opp-sex partner",
+            "Same-sex spouse",
+            "Same-sex partner",
+        ]
+
+        # Index helpers
+        # Mother index groups
+        mother_rp_idx = mothers_households.loc[
+            mothers_households["mother_relation_to_household_head"] == "Reference person"
+        ].index
+        mother_partner_idx = mothers_households.loc[
+            mothers_households["mother_relation_to_household_head"].isin(partners)
+        ].index
+        # Potential partner index groups
+        partners_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"].isin(partners)
+        ].index
+        reference_person_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"] == "Reference person"
+        ].index
+
+        # Assign second guardian to random partner of mothers
+        partner_ids = mothers_households.loc[mother_rp_idx.intersection(partners_idx)]
+        if len(partner_ids) > 0:
+            partner_ids = self.choose_random_guardian(partner_ids, "mother_id")
+            new_births.loc[
+                new_births["parent_id"].isin(partner_ids.index), "guardian_2"
+            ] = new_births["parent_id"].map(partner_ids)
+
+        reference_person_ids = mothers_households.loc[
+            mother_partner_idx.intersection(reference_person_idx)
+        ]
+        if len(reference_person_ids) > 0:
+            reference_person_ids = self.choose_random_guardian(
+                reference_person_ids, "mother_id"
+            )
+            new_births.loc[
+                new_births["parent_id"].isin(reference_person_ids.index), "guardian_2"
+            ] = new_births["parent_id"].map(reference_person_ids)
+
+        return new_births
+
+    @staticmethod
```

Comment:
Pycharm encourages you to make methods static if they can be. This function doesn't need access to the `self` object so it can function as a static method. See here for some insights into the benefits - https://stackoverflow.com/questions/2438473/what-is-the-advantage-of-using-static-methods. In this case, it makes very little difference.